x-pack/plugins/file_data_visualizer/public/application/components/stats_table/components/field_data_expanded_row/choropleth_map.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useMemo } from 'react';
+import { EuiFlexItem, EuiSpacer, EuiText, htmlIdGenerator } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  FIELD_ORIGIN,
+  SOURCE_TYPES,
+  STYLE_TYPE,
+  COLOR_MAP_TYPE,
+} from '../../../../../../../maps/common/constants';
+import { EMSTermJoinConfig } from '../../../../../../../maps/public';
+import { FieldVisStats } from '../../types';
+import { VectorLayerDescriptor } from '../../../../../../../maps/common/descriptor_types';
+import { EmbeddedMapComponent } from '../../../embedded_map';
+
+export const getChoroplethTopValuesLayer = (
+  fieldName: string,
+  topValues: Array<{ key: any; doc_count: number }>,
+  { layerId, field }: EMSTermJoinConfig
+): VectorLayerDescriptor => {
+  return {
+    id: htmlIdGenerator()(),
+    label: i18n.translate('xpack.fileDataVisualizer.choroplethMap.topValuesCount', {
+      defaultMessage: 'Top values count for {fieldName}',
+      values: { fieldName },
+    }),
+    joins: [
+      {
+        // Left join is the id from the type of field (e.g. world_countries)
+        leftField: field,
+        right: {
+          id: 'anomaly_count',
+          type: SOURCE_TYPES.TABLE_SOURCE,
+          __rows: topValues,
+          __columns: [
+            {
+              name: 'key',
+              type: 'string',
+            },
+            {
+              name: 'doc_count',
+              type: 'number',
+            },
+          ],
+          // Right join/term is the field in the doc you’re trying to join it to (foreign key - e.g. US)
+          term: 'key',
+        },
+      },
+    ],
+    sourceDescriptor: {
+      type: 'EMS_FILE',
+      id: layerId,
+    },
+    style: {
+      type: 'VECTOR',
+      // @ts-ignore missing style properties. Remove once 'VectorLayerDescriptor' type is updated
+      properties: {
+        icon: { type: STYLE_TYPE.STATIC, options: { value: 'marker' } },
+        fillColor: {
+          type: STYLE_TYPE.DYNAMIC,
+          options: {
+            color: 'Blue to Red',
```

Comment:
Should this be internationalized?

Reply:
No, this is just for the embeddable map config - the user doesn't see it.